### PR TITLE
readonly abn field for existing sellers

### PR DESCRIPTION
**Pull request by maxious**

### CODE REVIEW

File: src/bundles/SellerRegistration/components/BusinessDetailsForm/BusinessDetailsForm.js [hunk 1]
```diff
@@ -89,7 +88,7 @@ class BusinessDetailsForm extends BaseForm {
                               (<span>You need an Australian Business Number to do business in Australia.&nbsp;
                               <a href='https://abr.gov.au/For-Business,-Super-funds---Charities/Applying-for-an-ABN/Apply-for-an-ABN/' target="_blank" rel="external">Apply for an ABN here.</a>
                           </span>)}
-                          disabled="disabled"
+                          readOnly={!!supplierCode}
```
Comment (matt-sm) [inline src/bundles/SellerRegistration/components/BusinessDetailsForm/BusinessDetailsForm.js]: is !! an actual operator?

File: src/shared/form/Textfield.js [hunk 2]
```diff
@@ -1,55 +1,63 @@
...
-      pattern={pattern}
-      maxLength={maxLength}
-      disabled={disabled}
-    />
-  </div>
-);
+class Textfield extends React.Component {
+    render() {
+        const {name, id, htmlFor, label, model, validators, messages, description, pattern, maxLength, disabled, readOnly, type} = this.props;
+        const readonly = (readOnly ? {"readOnly": true, "disabled": true} : {});
```
Comment (matt-sm) [inline src/shared/form/Textfield.js]: why not make the props we pass to our Textfield match what Control supports?

File: src/bundles/SellerRegistration/components/BusinessDetailsForm/BusinessDetailsForm.js (hunk 1 above)
Comment (maxious) [inline src/bundles/SellerRegistration/components/BusinessDetailsForm/BusinessDetailsForm.js]: yes, webpack told me to use it - !  converts the object being not null/null to true/false and then another negates that ie. readOnly is true if supplierCode is not null

Comment (matt-sm) [inline src/bundles/SellerRegistration/components/BusinessDetailsForm/BusinessDetailsForm.js]: cool - didn't know that!

File: src/shared/form/Textfield.js (hunk 2 above)
Comment (maxious) [inline src/shared/form/Textfield.js]: I've split disabled and readOnly so they match all the way through

**Pull request merged**